src/fswAlgorithms/imageProcessing/centerOfBrightness/_UnitTest/test_centerOfBrightness.py

```diff
@@ -0,0 +1,186 @@
+
+    # # Create test thread
+    testProcessRate = macros.sec2nano(0.5)     # update process rate update time
+    testProc = unitTestSim.CreateNewProcess(unitProcessName)
+    testProc.addTask(unitTestSim.CreateNewTask(unitTaskName, testProcessRate))
+
+
+    # Construct algorithm and associated C++ container
+    moduleConfig = centerOfBrightness.CenterOfBrightness()
+    moduleConfig.ModelTag = "centerOfBrightness"
+
+    # Add test module to runtime call list
+    unitTestSim.AddModelToTask(unitTaskName, moduleConfig)
+
+    moduleConfig.filename = imagePath
+    moduleConfig.blurSize = blur
+    moduleConfig.threshold = 50
+    moduleConfig.saveDir = path + '/result_save.png'
+    if saveTest:
+        moduleConfig.saveImages = 1
+    cob_ref = [input_image.width/2, input_image.height/2]
+    if image == "half_half.png":
+        cob_ref = [(3*input_image.width/4 + 1)*valid, int(input_image.height/2)*valid]
+        pixelNum_ref = (int(input_image.width*input_image.height/2) + input_image.width)*valid
+
+    # Create input message and size it because the regular creator of that message is not part of the test.
+    inputMessageData = messaging.CameraImageMsgPayload()
+    inputMessageData.timeTag = int(1E9)
+    inputMessageData.cameraID = 1
+    inputMessageData.valid = valid
+    imgInMsg = messaging.CameraImageMsg().write(inputMessageData)
+    moduleConfig.imageInMsg.subscribeTo(imgInMsg)
+
+    # Setup logging on the test module output message so that we get all the writes to it
+    dataLog = moduleConfig.opnavCOBOutMsg.recorder()
+    unitTestSim.AddModelToTask(unitTaskName, dataLog)
+
+    # Need to call the self-init and cross-init methods
+    unitTestSim.InitializeSimulation()
+
+    # Set the simulation time.
+    # NOTE: the total simulation time may be longer than this value. The
+    # simulation is stopped at the next logging event on or after the
+    # simulation end time.
+    unitTestSim.ConfigureStopTime(macros.sec2nano(2.0))        # seconds to stop simulation
+
+    # Begin the simulation time run set above
+    unitTestSim.ExecuteSimulation()
+
+    center = dataLog.centerOfBrightness[-1,:]
+    pixelNum = dataLog.pixelsFound[-1]
+
+    # Output image:
+    output_image = Image.new("RGB", input_image.size)
+    output_image.paste(input_image)
+    draw_result = ImageDraw.Draw(output_image)
+
+    imageProcCircles = []
+    if pixelNum > 0:
+        imageProcCircles.append((center[0], center[1], np.sqrt(pixelNum)/50))
+    for x, y, r in imageProcCircles:
+        draw_result.ellipse((x - r, y - r, x + r, y + r), outline=(255, 0, 0, 0))
+
+    # Remove saved images for the test of that functionality
+    files = glob.glob(path + "/result_*")
+    for f in files:
+        os.remove(f)
+    # Save output image
+    if saveImage:
+        output_image.save("result_" + image)
+
+    if show_plots:
+        output_image.show()
+
+    #   print out success message if no error were found
```

Comment:
:large_orange_diamond: This comment seems wrong. Did you mean "failure" rather than "success"?

Reply:
Removed